updated javaScript-sliceMethod-issue#4378 (#4383)
Codecademy/docs: content/javascript/concepts/arrays/terms/slice/slice.md

---
Title: '.slice()'

> Description: 'Creates a partial copy of an array without altering the original array.'

Reviewer: ```suggestion
Description: 'Returns a shallow copy of a part of an array without altering the original array.'
```

Subjects:
  - 'Web Development'
  - 'Computer Science'
Tags:
  - 'Arrays'
  - 'Methods'

>   - 'Elements'

Reviewer: Please remove this tag, it is not necessary.

CatalogContent:
  - 'introduction-to-javascript'
  - 'paths/front-end-engineer-career-path'
---

> The **`.slice()`** [method](https://www.codecademy.com/resources/docs/javascript/methods) in JavaScript is utilized to select a segment of an [array](https://www.codecademy.com/resources/docs/javascript/arrays) and generate a new array from that selection. Importantly, this method creates this partial copy of an array, otherwise known as a _shallow copy_, without altering the original array.

Reviewer: ```suggestion
The **`.slice()`** [method](https://www.codecademy.com/resources/docs/javascript/methods) in JavaScript returns a partial copy of an array, otherwise known as a shallow copy, without altering the original array.
```

I have suggested to remove some redundant sentences. Also note that linking the linking array doesn't make sense as this entry will anyway be inside of the array concept.

## Understanding Shallow Copy

When the `.slice()` method is used, a new array known as a shallow copy is generated. A shallow copy can be imagined as a photo of the original array. This photo (the new array) looks exactly like the original, but it's a separate entity.

The key part is that, if the original array has primitive type elements like numbers or [strings](https://www.codecademy.com/resources/docs/javascript/strings), then changing these items in the new array or the shallow copy won't affect the original array at all, just as modifying a photo doesn't change the original scene.

> However, if the original array contains [objects](https://www.codecademy.com/resources/docs/javascript/objects) or arrays, then the shallow copy will reference them rather than duplicating them. Consequently, modifications to the elements in the shallow copy will impact the original array's elements. This happens because both the original and the new array are linked to the same elements, not separate copies of them.

Reviewer: You need to remove this section. The focus of this entry should be on `.slice` method. I understand that it is necessary to explain what a shallow copy is, so I suggest you condense this into 1 or 2 sentences and add that before syntax, without creating a separate subheading.
